bin/oneoff/data_fix/fix_invalid_users

```diff
@@ -54,18 +54,18 @@ generate_username_errors = 0
 User.where(username: nil).
   where.not(provider: nil).
   find_in_batches(start: START_USER_ID, finish: END_USER_ID, batch_size: BATCH_SIZE) do |user_batch|
-  ActiveRecord::Base.transaction do
-    user_batch.each do |user|
+  user_batch.each do |user|
+    ActiveRecord::Base.transaction do
       usernames_generated += 1
       user.username = UserHelpers.generate_username(User.with_deleted, user.name)
       CDO.log.info "Generated username: id=#{user.id}, username=#{user.username}"
       user.save!
-    rescue Exception => e
-      generate_username_errors += 1
-      CDO.log.error e
-    end
 
-    raise ActiveRecord::Rollback unless options[:actually_update]
+      raise ActiveRecord::Rollback unless options[:actually_update]
+    end
+  rescue Exception => e
```

Comment:
Rescuing Exception can have [unanticipated behavior](https://thoughtbot.com/blog/rescue-standarderror-not-exception).
```suggestion
  rescue StandardError => e
```

Reply:
Thanks for catching this!  I've changed this to `rescue => ex` (which catches StandardError).

Reply:
> Rescuing Exception can cause unanticipated behavior.

😬  

![Screen Shot 2020-12-04 at 11 14 15 AM](https://user-images.githubusercontent.com/8001765/101205054-0a07bf80-3622-11eb-91c8-8fc7eb218f84.png)

Reply:
We should add a linter rule for this :(

Reply:
Looks like there’s one we can enable https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/RescueException